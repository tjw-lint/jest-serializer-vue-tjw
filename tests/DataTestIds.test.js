const helpers = require('./helpers.js');

import { shallowMount } from '@vue/test-utils';
import DataTestIds from './components/DataTestIds.vue';

describe('DataTestIds.vue', () => {
  test('Remove data-server-rendered, data-test, data-testid, data-test-id by default', () => {
    helpers.mockSettings({});

    const wrapper = shallowMount(DataTestIds);

    expect(wrapper)
      .toMatchSnapshot();
  });

  test('Only data-server-rendered removed', () => {
    helpers.mockSettings({
      removeServerRendered: true,
      removeDataTest: false,
      removeDataTestid: false,
      removeDataTestId: false,
      removeDataQa: false
    });

    const wrapper = shallowMount(DataTestIds);

    expect(wrapper)
      .toMatchSnapshot();
  });

  test('Only data-test removed', () => {
    helpers.mockSettings({
      removeServerRendered: false,
      removeDataTest: true,
      removeDataTestid: false,
      removeDataTestId: false,
      removeDataQa: false
    });

    const wrapper = shallowMount(DataTestIds);

    expect(wrapper)
      .toMatchSnapshot();
  });

  test('Only data-testid removed', () => {
    helpers.mockSettings({
      removeServerRendered: false,
      removeDataTest: false,
      removeDataTestid: true,
      removeDataTestId: false,
      removeDataQa: false
    });

    const wrapper = shallowMount(DataTestIds);

    expect(wrapper)
      .toMatchSnapshot();
  });

  test('Only data-test-id removed', () => {
    helpers.mockSettings({
      removeServerRendered: false,
      removeDataTest: false,
      removeDataTestid: false,
      removeDataTestId: true,
      removeDataQa: false
    });

    const wrapper = shallowMount(DataTestIds);

    expect(wrapper)
      .toMatchSnapshot();
  });

  test('Only data-qa removed', () => {
    helpers.mockSettings({
      removeServerRendered: false,
      removeDataTest: false,
      removeDataTestid: false,
      removeDataTestId: false,
      removeDataQa: true
    });

    const wrapper = shallowMount(DataTestIds);

    expect(wrapper)
      .toMatchSnapshot();
  });
});
