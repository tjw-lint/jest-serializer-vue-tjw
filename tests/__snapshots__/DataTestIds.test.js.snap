// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTestIds.vue Only data-qa removed 1`] = `
<ul>
  <li data-server-rendered="true">
    data-server-rendered
  </li>
  <li data-test="data-test">
    data-test
  </li>
  <li data-testid="data-testid">
    data-testid
  </li>
  <li data-test-id="data-test-id">
    data-test-id
  </li>
  <li>
    data-qa
  </li>
</ul>
`;

exports[`DataTestIds.vue Only data-server-rendered removed 1`] = `
<ul>
  <li>
    data-server-rendered
  </li>
  <li data-test="data-test">
    data-test
  </li>
  <li data-testid="data-testid">
    data-testid
  </li>
  <li data-test-id="data-test-id">
    data-test-id
  </li>
  <li data-qa="data-qa">
    data-qa
  </li>
</ul>
`;

exports[`DataTestIds.vue Only data-test removed 1`] = `
<ul>
  <li data-server-rendered="true">
    data-server-rendered
  </li>
  <li>
    data-test
  </li>
  <li data-testid="data-testid">
    data-testid
  </li>
  <li data-test-id="data-test-id">
    data-test-id
  </li>
  <li data-qa="data-qa">
    data-qa
  </li>
</ul>
`;

exports[`DataTestIds.vue Only data-test-id removed 1`] = `
<ul>
  <li data-server-rendered="true">
    data-server-rendered
  </li>
  <li data-test="data-test">
    data-test
  </li>
  <li data-testid="data-testid">
    data-testid
  </li>
  <li>
    data-test-id
  </li>
  <li data-qa="data-qa">
    data-qa
  </li>
</ul>
`;

exports[`DataTestIds.vue Only data-testid removed 1`] = `
<ul>
  <li data-server-rendered="true">
    data-server-rendered
  </li>
  <li data-test="data-test">
    data-test
  </li>
  <li>
    data-testid
  </li>
  <li data-test-id="data-test-id">
    data-test-id
  </li>
  <li data-qa="data-qa">
    data-qa
  </li>
</ul>
`;

exports[`DataTestIds.vue Remove data-server-rendered, data-test, data-testid, data-test-id by default 1`] = `
<ul>
  <li>
    data-server-rendered
  </li>
  <li>
    data-test
  </li>
  <li>
    data-testid
  </li>
  <li>
    data-test-id
  </li>
  <li data-qa="data-qa">
    data-qa
  </li>
</ul>
`;
